src/receive_imf.rs

```diff
@@ -2946,7 +2996,8 @@ async fn add_or_lookup_contacts_by_address_list(
     address_list: &[SingleInfo],
     origin: Origin,
 ) -> Result<Vec<ContactId>> {
-    let mut contact_ids = HashSet::new();
+    let mut contact_ids = Vec::new();
+    let mut contact_ids_set = HashSet::new();
```

Comment:
Add comment(non-blocking): I guess that this change is so that the order of the contact ids is preserved? It took me some time to understand this, so, would be nice to add a comment. E.g.:

```suggestion
    // In order to return the contact ids in the same order as they
    // were in the address_list, but still deduplicate them efficiently,
    // collect them in both a Vec and a HashSet:
    let mut contact_ids_set = HashSet::new();
```

(Alternatively, only use the Vec and then deduplicate them, e.g.:
```rust
    // Deduplicate the contact ids:
    let mut set = HashSet::new(); 
    contact_ids.retain(|id| set.insert(*id));
```
)